enterprise/coderd/features_internal_test.go

```diff
@@ -1,545 +0,0 @@
-			entitlements: entitlements{
-				hasLicense: true,
-				activeUsers: numericalEntitlement{
-					entitlement{entitled},
-					entitlementLimit{
-						unlimited: false,
-						limit:     100,
-					},
-				},
-				auditLogs: entitlement{entitled},
-			},
-		}
-		_, err := db.InsertUser(ctx, database.InsertUserParams{
-			ID:             uuid.UUID{},
-			Email:          "",
-			Username:       "",
-			HashedPassword: nil,
-			CreatedAt:      time.Time{},
-			UpdatedAt:      time.Time{},
-			RBACRoles:      nil,
-			LoginType:      "",
-		})
-		require.NoError(t, err)
-		result := requestEntitlements(t, uut)
-		assert.True(t, result.HasLicense)
-		ul := result.Features[codersdk.FeatureUserLimit]
-		assert.Equal(t, codersdk.EntitlementEntitled, ul.Entitlement)
-		assert.Equal(t, int64(100), *ul.Limit)
-		assert.Equal(t, int64(1), *ul.Actual)
-		assert.True(t, ul.Enabled)
-		al := result.Features[codersdk.FeatureAuditLog]
-		assert.Equal(t, codersdk.EntitlementEntitled, al.Entitlement)
-		assert.True(t, al.Enabled)
-		assert.Nil(t, al.Limit)
-		assert.Nil(t, al.Actual)
-		assert.Empty(t, result.Warnings)
-	})
-
-	t.Run("Warnings", func(t *testing.T) {
-		t.Parallel()
-		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitShort)
-		defer cancel()
-		db := databasefake.New()
-		uut := &featuresService{
-			logger:      logger,
-			database:    db,
-			pubsub:      pubsub,
-			keys:        map[string]ed25519.PublicKey{keyID: pub},
-			enablements: Enablements{AuditLogs: true},
-			entitlements: entitlements{
-				hasLicense: true,
-				activeUsers: numericalEntitlement{
-					entitlement{gracePeriod},
-					entitlementLimit{
-						unlimited: false,
-						limit:     4,
-					},
-				},
-				auditLogs: entitlement{gracePeriod},
-			},
-		}
-		for i := byte(0); i < 5; i++ {
-			_, err := db.InsertUser(ctx, database.InsertUserParams{
-				ID:             uuid.UUID{i},
-				Email:          "",
-				Username:       "",
-				HashedPassword: nil,
-				CreatedAt:      time.Time{},
-				UpdatedAt:      time.Time{},
-				RBACRoles:      nil,
-				LoginType:      "",
-			})
-			require.NoError(t, err)
-		}
-		result := requestEntitlements(t, uut)
-		assert.True(t, result.HasLicense)
-		ul := result.Features[codersdk.FeatureUserLimit]
-		assert.Equal(t, codersdk.EntitlementGracePeriod, ul.Entitlement)
-		assert.Equal(t, int64(4), *ul.Limit)
-		assert.Equal(t, int64(5), *ul.Actual)
-		assert.True(t, ul.Enabled)
-		al := result.Features[codersdk.FeatureAuditLog]
-		assert.Equal(t, codersdk.EntitlementGracePeriod, al.Entitlement)
-		assert.True(t, al.Enabled)
-		assert.Nil(t, al.Limit)
-		assert.Nil(t, al.Actual)
-		assert.Len(t, result.Warnings, 2)
-		assert.Contains(t, result.Warnings,
-			"Your deployment has 5 active users but is only licensed for 4.")
-		assert.Contains(t, result.Warnings,
-			"Audit logging is enabled but your license for this feature is expired.")
-	})
-}
-
-func TestFeaturesServiceSyncEntitlements(t *testing.T) {
```

Comment:
I can't see where these tests got moved.  Did they just get deleted?  We need some tests of the license synchronization

Reply:
Yup, since this was in draft I wanted your thoughts before I'd continue refactoring these tests. They'll be added back once we resolve comments further up.